pages/science/reactors/extensiblereactor.rst

```diff
@@ -0,0 +1,181 @@
+allows the user to modify the governing equations corresponding to 
+the following state variables:
+
+- :math:`m`, the mass of the reactor's contents (in kg)
+
+- :math:`T`, the temperature (in K)
+
+- :math:`Y_k`, the mass fractions for each species (dimensionless)
+
+As shown in the derivations of the governing equations for an Ideal Gas 
+Constant Pressure Reactor, the user may modify the 3 equations below:
+
+.. math::
+
+   \frac{dm}{dt} = \sum_{in} \dot{m}_{in} - \sum_{out} \dot{m}_{out} +
+                    \dot{m}_{wall}
+                    \tag{1}
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+.. math::
+
+   m \frac{dY_k}{dt} = \sum_{in} \dot{m}_{in} (Y_{k,in} - Y_k)+
+                      \dot{m}_{k,gen} - Y_k \dot{m}_{wall}
+                      \tag{3}
+
+There are two "sides" to each of these equations: the terms left of the equals
+sign and the terms to the right of the equals sign. This is the format
+in which the user will be editing the governing equations. For example,
+if the user wishes to add a term for a large mass (say a rock) inside
+the reactor to see the effects on reaction temperature:
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+Will change to:
+
+.. math::
+
+   m c_p \frac{dT}{dt} + m_{rock} c_{p,rock} \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+
+A simple example is shown below to illustrate the process for implementing
+changes in Cantera's existing governing equations.
+We will be replacing the right-hand side (RHS) and left-hand side (LHS) of 
+the temperature governing equation for an Ideal Gas Constant Pressure Reactor.
+All other governing equations defining an Ideal Gas Constant Pressure Reactor
+will remain as the default.
+
+In this example
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+Will change to:
+
+.. math::
+
+   m_{rock} c_{p,rock}\frac{dT}{dt} + m_{gas}\frac{dT}{dt} = - \dot{Q}
+
+The governing equations will be modified through the user created Python class' methods.
+For each method, the name should be prefixed with **before_**, **after_**, or 
+**replace_**, indicating whether the this method should be called before, after, 
+or instead of the corresponding method from the base class.
+
+.. code-block:: python
+
+  #1 Define objects, properties, and initial conditions.
+
+   #create gas object
+   gas = ct.Solution('h2o2.yaml')
+   gas.TPX = 500, ct.one_atm, 'H2:2,O2:1,N2:4'
+
+   #define properties of gas and solid
+   mass_gas = 20 #[kg]
+   Q = 100 #[J/s]
+   mass_rock = 10 #[kg]
+   cp_rock = 1.0 #[J/kgK]
```

Comment:
This is kind of pedantic, but it's worth it for code that we supply as examples. PEP8 suggests:
- Each line of a block comment starts with a # and a single space (unless it is indented text inside the comment).
- Inline comments should be separated by at least two spaces from the statement. They should start with a # and a single space.
So this should be:
```suggestion
   # define properties of gas and solid
   mass_gas = 20  # [kg]
   Q = 100  # [J/s]
   mass_rock = 10  # [kg]
   cp_rock = 1.0  # [J/kgK]
```